_posts/2022-08-24-#25768.md

```diff
@@ -7,18 +7,101 @@ authors: [achow101]
+the node's current (mockable) time and the wallet's randomly chosen `nNextResend` time to decide
+whether or not it's time to call `SubmitTxMemoryPoolAndRelay()`. As such, to trigger a rebroadcast,
+the clock must pass two time points: `CWallet::nNextResend` and the time at which the `scheduler`
+has `MaybeResendWalletTxs()` task set for.
+
+- [PR #25768](https://github.com/bitcoin/bitcoin/pull/25768) addresses an issue in which the wallet
+  rebroadcasts multiple transactions with dependency relationships. It must submit a parent
+transaction before its child, otherwise the mempool will reject it due to missing inputs.
+
+- Prior to this PR, the transactions are rebroadcast in whichever order they appear in
+  [`CWallet::mapWallet`](https://github.com/bitcoin/bitcoin/blob/d480586ecb1d4c265ffec278f10c46500d89382f/src/wallet/wallet.h#L394-L396),
+which is a `std::unordered_map`.
+
+- This PR reuses the logic in `ReacceptWalletTransactions()` to sort `mapWallet` transactions by
+  [`CWalletTx::nOrderPos`](https://github.com/bitcoin/bitcoin/blob/d480586ecb1d4c265ffec278f10c46500d89382f/src/wallet/transaction.h#L185).
+
 ## Questions
-1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
 
+1. Did you review the PR? What was your review approach?
+
+1. Did you leave a [Concept ACK, approach ACK, tested ACK, or
+   NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review) on the PR?
+
+### Concept
+
+1. What incorrect behavior does this PR address?
+
+1. Were you able to reproduce the issue? Does the test added in the PR adequately test this
+   behavior, i.e., does it succeed with this fix but fail without it?
+
+1. Why is it difficult to write a test that reliably demonstrates that transactions may be
+   rebroadcast in the wrong order? How does `std::unordered_map` order its elements?
+
+### Approach
+
+1. Can you think of any other methods of getting the wallet transactions sorted in this order?
+
+1. Would it make sense to *keep* the wallet transactions sorted in this order? Why is `mapWallet`
+   implemented as a `std::unordered_map`; would another data structure make sense?
+
+1. If you thought of an alternative approach, how does it compare? What is its runtime complexity?
+   How much {more, less} memory is allocated? Does it require a lot of code to be
+(re)written? Is its complexity appropriate for its purpose?
+
+### Implementation
+
+1. How is `GetSortedTxs()`, the function extracted from `ReacceptWalletTransactions()`, implemented?
+   How much memory is needed, as a function of the transactions in `mapWallet`?
+
+1. `GetSortedTxs()` is
+   [declared](https://github.com/bitcoin-core-review-club/bitcoin/commit/9a556564e9dc64ae0ad723c78da33d0c982f006f#diff-9ce137cd784ea308778842120aa2af6d2bb8369485b71f25e72b2a32cf0a5b21R536)
+as requiring the wallet lock to already be held. Why or why not does this make sense?
```

Comment:
```suggestion
as requiring the wallet lock to already be held. Why does this (not) make sense?
```

Reply:
I prefer "why or why not" tbh

Reply:
Okay then keep it. I feel like "Why or why not?" works well in a separate sentence but "why not does this" sounds a bit weird to me. But no strong feelings about it so whatever you prefer - not worth spending too much time on!

Reply:
Yeah makes sense, thanks for the feedback! Mostly did it habitually, hopefully it's not too confusing for people heh.